src/c_api/value.cpp

```diff
@@ -182,388 +182,883 @@ kuzu_value* kuzu_value_clone(kuzu_value* value) {
     return c_value;
 }
 
-void kuzu_value_copy(kuzu_value* value, kuzu_value* other) {
+kuzu_state kuzu_value_copy(kuzu_value* value, kuzu_value* other) {
```

Comment:
I would double check if `copyValueFrom` can thorw error? I think this cannot fail? If not, it would be better to make this `void`.